### ISSUE TITLE
Getting "pageName.trim is not a function"

### ISSUE THREAD

**Issue by caiw** [jsepia/markdown-it-wikilinks#16]
I'm using this as a plugin for Eleventy. I have a wikilink `[[contact]]` in one of my pages (`404.md`). If I try to build I get the error:

```
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] 1. Having trouble rendering njk template ./source/pages/404.md (via TemplateContentRenderError)
[11ty] 2. pageName.trim is not a function (via TypeError)
[11ty] 
[11ty] Original error stack trace: TypeError: pageName.trim is not a function
[11ty]     at Object.postProcessPageName (/Users/cai/Dox/Creative/Web design/cai.zone-11ty/node_modules/markdown-it-wikilinks/index.js:21:27)
...
```

If I put in a `console.log(pageName);` within `postProcessName`, I get this printing: `[ 'contact' ]`, so I guess it's complaining because the array `['contact']` has no `.trim()`.

However, if I replace my wikilink by `[[contact|contact]]`, I the build goes through fine and I get ``contact`` in the log (and the link works too!).

I'm afraid I don't know what I'm doing well enough to track it down further than that...

**Comment by jsepia** [comment]
Thanks for the report!

I'm not sure why your `postProcess*` functions are receiving arrays instead of plain strings. I wrote the following test and was unable to reproduce the error:

```javascript
const TEST_PAGE_NAME = 'contact';
const TEST_MARKDOWN = '[[contact]]';
const EXPECTED_OUTPUT = '<a href="./contact.html">contact</a>';

function execute(postProcessCallbackName) {
  const cb = sinon.stub();
  cb.returns(TEST_PAGE_NAME);
  const opts = {};
  opts[postProcessCallbackName] = cb;
  const plugin = MarkdownItWikilinks(opts)
  const md = require('markdown-it')()
         .use(plugin);
  const result = md.renderInline(TEST_MARKDOWN);
  assert.equal(result, EXPECTED_OUTPUT);
  assert.equal(cb.calledOnce, true);
  assert.equal(cb.lastCall.args[0], TEST_PAGE_NAME);
}

it('should pass the correct arguments to postProcessPagePath', () => {
  execute('postProcessPagePath'); // Success
});

it('should retain backwards compatibility with postProcessPageName', () => {
  execute('postProcessPageName'); // Success
});
```

Can you please provide the following information to help me troubleshoot the issue?

1. What version of `markdown-it-wikilinks` are you using?
2. By `postProcessName` do you mean `postProcessPageName`?
3. Please paste/share your code

Thanks

**Comment by caiw** [comment]
This is how I am invoking the plugin in my `.eleventy.js`.

```javascript
const mdWikilinks = require('markdown-it-wikilinks')({
	'uriSuffix': "/",
	'htmlAttributes': { 'class': 'wikilink' },
	'generatePageNameFromLabel': function(label) {
		return label.split("/").map(function (pathSegment) {
			pathSegment = _.deburr(pathSegment);
			pathSegment = pathSegment.replace(/[^\w\s]/g, '-');
			pathSegment = pathSegment.toLowerCase();
			return pathSegment;
		});
	}
});
var md = require('markdown-it')({
  html: true,
  linkify: false,
  typographer: true,
}).use(mdFootnotes)
  .use(mdWikilinks);
```

I have found that if I remove my own `generatePageNameFromLabel`, it builds ok, i.e.:

```javascript
const mdWikilinks = require('markdown-it-wikilinks')({
	'uriSuffix': "/",
	'htmlAttributes': { 'class': 'wikilink' }
});
var md = require('markdown-it')({
  html: true,
  linkify: false,
  typographer: true,
}).use(mdFootnotes)
  .use(mdWikilinks);
```
is fine.  So perhaps I have some issue in this function?

Sorry I'm not able to produce a self-contained reproduction yet, I will if I can.

**Comment by jsepia** [comment]
> ```js
> 	'generatePageNameFromLabel': function(label) {
> 		return label.split("/").map(function (pathSegment) {
> 			pathSegment = _.deburr(pathSegment);
> 			pathSegment = pathSegment.replace(/[^\w\s]/g, '-');
> 			pathSegment = pathSegment.toLowerCase();
> 			return pathSegment;
> 		});
> 	}
> ```

I think this is the issue. This function `split`s the label value but doesn't `join` it back again. In other words you're missing something like this before the return statement:

```js
pathSegment = pathSegment.join('-');
```

**Comment by caiw** [comment]
Ah hah, thank you!  This worked for me:

```js
'generatePageNameFromLabel': function(label) {
	return label.split("/").map(function (pathSegment) {
		pathSegment = _.deburr(pathSegment);
		pathSegment = pathSegment.replace(/[^\w\s]/g, '-');
		pathSegment = pathSegment.toLowerCase();
		return pathSegment;
	}).join("/");  // <-- this is the new line
},
```

I actually modified the original based on the [example in the docs](https://github.com/jsepia/markdown-it-wikilinks#example), so I wonder if the same `.join` is missing from that too? :)

Thanks again for your help.

**Comment by jsepia** [comment]
> I actually modified the original based on the [example in the docs](https://github.com/jsepia/markdown-it-wikilinks#example), so I wonder if the same `.join` is missing from that too? :)

Oh you're right! Sorry, I wrote that example such a long time ago. Fixed in 48641e9

I'm closing this issue for now. Feel free to reopen it if needed.
